s3/s3.go

```diff
@@ -0,0 +1,243 @@
+	aws_secret_access_key := resp.Credentials.SecretAccessKey
+	aws_session_token := resp.Credentials.SessionToken
+
+	creds := credentials.NewStaticCredentials(aws_access_key_id, aws_secret_access_key, aws_session_token)
+	_, err = creds.Get()
+	if err != nil {
+		fmt.Printf("Invalid credentials: %s", err)
+		return false
+	}
+
+	// Get bucket location
+	aws_config := aws.NewConfig().WithCredentials(creds).WithRegion("us-west-2")
+	sess := session.New(aws_config)
+	svc := awsS3.New(sess)
+	params := &awsS3.GetBucketLocationInput{
+		Bucket: aws.String(bucket),
+	}
+	res, err := svc.GetBucketLocation(params)
+	if err != nil {
+		fmt.Println(err.Error())
+		return false
+	}
+	region := res.LocationConstraint
+
+	// Create a session which contains the configurations for the SDK.
+	// Use the session to create the service clients to make API calls to AWS.
+	aws_config = aws.NewConfig().WithCredentials(creds).WithRegion(*region)
+	sess = session.New(aws_config)
+	svc = awsS3.New(sess)
+
+	if argv["put"] == true {
+		upload := Put(svc, filename, bucket, prefix)
+		return upload
+	} else if argv["get"] == true {
+		download := Get(svc, key, bucket, prefix)
+		return download
+	} else {
+		panic(fmt.Sprint("Invalid subcommand."))
+	}
+}
+
+func Put(svc *awsS3.S3, filename string, bucket string, prefix string) bool {
+	// Open the filename and return the file
+	file, err := os.Open(filename)
+	if err != nil {
+		fmt.Println("Failed to open file", filename, err)
+		return false
+	}
+	defer file.Close()
+
+	// Generate key from filename
+	key := filepath.Base(filename)
+
+	fmt.Printf("Uploading %s to s3://%s/%s...\n", key, bucket, prefix)
+
+	// Upload the file
+	params := &awsS3.PutObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(prefix + key),
+		Body:   file,
+	}
+	_, err = svc.PutObject(params)
+	if err != nil {
+		fmt.Println(err.Error())
+		return false
+	}
+
+	fmt.Printf("Successfully uploaded %s to s3://%s/%s\n", key, bucket, prefix)
+	return true
+}
+
+func Get(svc *awsS3.S3, key string, bucket string, prefix string) bool {
+	// Get current path
+	pwd, err := os.Getwd()
+	if err != nil {
+		fmt.Println("Error occured: ", err)
+		return false
+	}
+
+	// Create filepath
+	filename := filepath.Join(pwd, key)
+	if err := os.MkdirAll(filepath.Dir(filename), 0775); err != nil {
+		fmt.Println("Unable to create directory: ", err)
+		return false
+	}
+
+	// Setup the local file
+	file, err := os.Create(filename)
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	fmt.Printf("Downloading from s3://%s/%s to %s...\n", bucket, prefix+key, filename)
+
+	// Download the file
+	params := &awsS3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(prefix + key),
+	}
+
+	resp, err := svc.GetObject(params)
+	if err != nil {
+		fmt.Printf("Failed to download data to %s from s3://%s/%s\n", filename, bucket, prefix+key)
+		return false
+	}
+
+	// Write data to the file
+	data, err := ioutil.ReadAll(resp.Body)
```

Comment:
This isn't smart if the body has a non-trivial size... a few GBs and this won't be fun...

`io.Copy(file, resp.Body)`

Also won't we need a retry loop here?